Assets/MRTK/Services/InputSystem/GazeProvider.cs

```diff
@@ -550,6 +549,12 @@ private async void RaiseSourceDetected()
         public void UpdateGazeInfoFromHit(MixedRealityRaycastHit raycastHit)
         {
             HitInfo = raycastHit;
+
+            if (IsEyeTrackingEnabledAndValid)
+            {
+                UpdateEyeGaze(null, GazePointer.Rays[0], DateTime.UtcNow);
```

Comment:
What's this change doing? `UpdateEyeGaze` should really only be called by an eye gaze provider, which I think is somewhat a requirement for `IsEyeTrackingEnabledAndValid` to be true. This seems like it's erasing the `IMixedRealityEyeGazeDataProvider` parameter and updating the time, but will the ray be any different?

Reply:
Update eye gaze doesn't seem to actually use the IMixedRealityEyeGazeDataProvider parameter

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/2c363bd001736fd58e5c4a7de33464a6269a6103/Assets/MRTK/Services/InputSystem/GazeProvider.cs#L614-L620

And looking at the other instances where UpdateEyeGaze is actually used, those classes also don't seem to have a real implementation of the IMixedRealityEyeGazeDataProvider.

https://github.com/microsoft/MixedRealityToolkit-Unity/blob/7421a2fbed35ac029438a1cf2ddeb65a994e76cf/Assets/MRTK/Core/Providers/InputSimulation/InputPlaybackService.cs#L95-L97

I'm a bit unsure what exactly IMixedRealityEyeSaccadeProvider refers to

Reply:
Oh, wild! Looks like it did in a very early implementation, but we never cleaned up the method header before shipping :( might be time to deprecate this one and add one without the ignored parameter?

Our main `IMixedRealityEyeSaccadeProvider` is https://github.com/microsoft/MixedRealityToolkit-Unity/blob/main/Assets/MRTK/Core/Utilities/EyeGazeSmoother.cs

Reply:
We'll definitely need to rework/rethink this interface as we move forward to MRTK v3!